type-tests/tsconfig.json

```diff
@@ -0,0 +1,23 @@
+{
+  "compilerOptions": {
+    "module": "commonjs",
+    "lib": ["es6", "dom"],
+    "noImplicitAny": true,
+    "noImplicitThis": true,
+    "strictNullChecks": true,
+    "strictFunctionTypes": true,
+    "noEmit": true,
+    "allowSyntheticDefaultImports": true,
```

Comment:
This indicates that all consumers also need this flag. More info here: https://github.com/ember-cli/ember-ajax/pull/441#discussion_r268309448